bridge-ui/src/utils/history/fetchETHBridgeEvents.ts

```diff
@@ -0,0 +1,122 @@
+import { config as wagmiConfig } from "@/lib/wagmi";
+import { defaultTokensConfig, HistoryActionsForCompleteTxCaching } from "@/stores";
+import {
+  BridgeTransaction,
+  BridgeTransactionType,
+  Chain,
+  ChainLayer,
+  Token,
+  MessageSentABIEvent,
+  MessageSentLogEvent,
+} from "@/types";
+import { formatOnChainMessageStatus } from "./formatOnChainMessageStatus";
+import { getCompleteTxStoreKey } from "./getCompleteTxStoreKey";
+
+export async function fetchETHBridgeEvents(
+  historyStoreActions: HistoryActionsForCompleteTxCaching,
+  lineaSDK: LineaSDK,
+  address: Address,
+  fromChain: Chain,
+  toChain: Chain,
+  tokens: Token[],
+): Promise<BridgeTransaction[]> {
+  const transactionsMap = new Map<string, BridgeTransaction>();
+
+  const client = getPublicClient(wagmiConfig, {
+    chainId: fromChain.id,
+  });
+
+  const contract = fromChain.layer === ChainLayer.L1 ? lineaSDK.getL2Contract() : lineaSDK.getL1Contract();
+
+  const messageServiceAddress = fromChain.messageServiceAddress;
+  const [ethLogsForSender, ethLogsForRecipient] = await Promise.all([
+    client.getLogs({
+      event: MessageSentABIEvent,
+      // No need to find more 'optimal' value than earliest.
+      // Empirical testing showed no practical difference when using hardcoded block number (that was 90 days old).
+      fromBlock: "earliest",
+      toBlock: "latest",
+      address: messageServiceAddress,
+      args: {
+        _from: address,
+      },
+    }),
+    client.getLogs({
+      event: MessageSentABIEvent,
+      fromBlock: "earliest",
+      toBlock: "latest",
+      address: messageServiceAddress,
+      args: {
+        _to: address,
+      },
+    }),
+  ]);
+
+  const messageSentLogs = [...ethLogsForSender, ...ethLogsForRecipient] as unknown as MessageSentLogEvent[];
+
+  const uniqueLogsMap = new Map<string, (typeof messageSentLogs)[0]>();
+  for (const log of messageSentLogs) {
+    const uniqueKey = `${log.args._from}-${log.args._to}-${log.transactionHash}`;
+    if (!uniqueLogsMap.has(uniqueKey)) {
+      uniqueLogsMap.set(uniqueKey, log);
+    }
+  }
+
+  const currentTimestamp = new Date();
+
+  await Promise.all(
+    Array.from(uniqueLogsMap.values()).map(async (log) => {
+      const uniqueKey = `${log.args._from}-${log.args._to}-${log.transactionHash}`;
+
+      // Search cache for completed tx for this txHash, if cache-hit can skip remaining logic
+      const cacheKey = getCompleteTxStoreKey(fromChain.id, log.transactionHash);
+      const cachedCompletedTx = historyStoreActions.getCompleteTx(cacheKey);
+      if (cachedCompletedTx) {
+        transactionsMap.set(uniqueKey, cachedCompletedTx);
+        return;
+      }
+
+      const messageHash = log.args._messageHash;
+
+      const block = await client.getBlock({ blockNumber: log.blockNumber, includeTransactions: false });
+
+      if (compareAsc(fromUnixTime(Number(block.timestamp.toString())), subDays(currentTimestamp, 90)) === -1) {
```

Comment:
the 90 is a magic number, please define it somewhere as a constant for ease of understanding

Reply:
have encapsulated this logic and constant in a new `isBlockTooOld` file